bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/SyncSupported.java

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client.api;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Marks Handles which support the 'sync' primitive
+ *
+ * @see WriteHandle
+ * @see WriteAdvHandle
+ *
+ * @since 4.6
+ */
+public interface SyncSupported {
+
+    /**
+     * Waits for all data written by this Handle to have been persisted durably on a quorum of bookies and advances
```

Comment:
this comment is confusing.

the behaviors of `sync` between `WriteHandle` and `WriteAdvHandle` are different. especially in `WriteAdvHandle`, you can't wait for all the data written to be persisted, because the entries might be added out of order.

Reply:
Yes,  WriteAdvHandle is tricky. You can leave gaps, then call sync() and then write entries in the gap.
In this case sync() can only guarantee that data sent to bookies "before calling sync" has been persisted.
Anyway the comment is right: "all data written by this Handle" are guaranteed to have been fsynched to the journal. And the returned value has no real meaning.
Maybe it is  better to have two separate methods, and for WriteAdvHandle return a CompletableFuture<Void>